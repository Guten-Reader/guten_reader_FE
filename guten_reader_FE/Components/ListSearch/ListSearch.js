import React from 'react';
import { StyleSheet, View, Button, Text } from 'react-native';

const ListSearch = ({ book }) => {

  const addBookToLibrary = async () => {
    //guten_id, author, title
    let newBook = {
      guten_id: book.id,
      title: book.title,
      author: book.author
    }
    console.log("New BOOK! ", newBook)
  }

  return (
    <View style={styles.container}>
      <View style={styles.text}>
        <Text style={styles.title}>{book.title}</Text>
        <Text style={styles.author}>{book.author}</Text>
      </View>
      <Button title="DOWNLOAD" onPress={addBookToLibrary} style={styles.button}/>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    marginBottom: 20,
    justifyContent: 'space-between'
  },
  title: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 20,
    paddingTop: 10,
    paddingBottom: 10,
    width: 200
  },
  author: {
    color: 'white',
    fontSize: 15,
    marginBottom: 10,
    width: 200
  }
});

export default ListSearch;
