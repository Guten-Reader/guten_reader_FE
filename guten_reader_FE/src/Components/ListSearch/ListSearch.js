import React, { Component } from 'react';
import { StyleSheet, View, Button, Text } from 'react-native';
import { addBook } from '../../apiCalls';

class ListSearch extends Component {
  constructor() {
    super();
  }

  addBookToLibrary = async () => {
    let newBook = {
      guten_id: this.props.book.id,
      title: this.props.book.title,
      author: this.props.book.author
    };
    addBook(1, newBook);
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.text}>
          <Text style={styles.title}>{this.props.book.title}</Text>
          <Text style={styles.author}>{this.props.book.author}</Text>
        </View>
        <Button title="DOWNLOAD" onPress={this.addBookToLibrary} style={styles.button}/>
        <Button title="CHECKOUT" style={styles.button}/>

      <Button title="CHECKOUT" style={styles.button}/>
    </View>
  )
}}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    marginBottom: 20,
    justifyContent: 'space-between'
  }, 
  title: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 20,
    paddingTop: 10,
    paddingBottom: 10,
    width: 200
  },
  author: {
    color: 'white',
    fontSize: 15,
    marginBottom: 10,
    width: 200
  }
});

export default ListSearch;
