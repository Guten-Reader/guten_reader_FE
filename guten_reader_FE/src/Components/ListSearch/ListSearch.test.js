import React from 'react';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ListSearch from './ListSearch';

describe('ListSearch renders correctly with Jest', () => {
  it('renders correctly', () => {
    const book = {
      title: 'This is a book',
      author: 'I am the author'
    }
    const wrapper = shallow(<ListSearch book={book} />);
    expect(wrapper).toMatchSnapshot();
  })
})