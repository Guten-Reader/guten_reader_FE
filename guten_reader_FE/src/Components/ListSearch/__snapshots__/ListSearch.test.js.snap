// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListSearch renders correctly with Jest renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 20,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "paddingBottom": 10,
          "paddingTop": 10,
          "width": 200,
        }
      }
    >
      This is a book
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 15,
          "marginBottom": 10,
          "width": 200,
        }
      }
    >
      I am the author
    </Text>
  </View>
  <Button
    title="DOWNLOAD"
  />
</View>
`;
